daemon/reload.go

```diff
@@ -308,3 +311,18 @@ func (daemon *Daemon) reloadLiveRestore(conf *config.Config, attributes map[stri
 	attributes["live-restore"] = fmt.Sprintf("%t", daemon.configStore.LiveRestoreEnabled)
 	return nil
 }
+
+// reloadNetworkDiagnose updates the network controller starting the diagnose mode if the config is valid
+func (daemon *Daemon) reloadNetworkDiagnose(conf *config.Config, attributes map[string]string) error {
+	if conf == nil || daemon.netController == nil {
+		return nil
+	}
+	// Enable the network diagnose if the flag is set with a valid port withing the range
+	if conf.IsValueSet("network-diagnose") && conf.NetworkDiagnose > 0 && conf.NetworkDiagnose < 65536 {
+		logrus.Infof("Calling the diagnose start with %d", conf.NetworkDiagnose)
```

Comment:
Wondering if we should make this a warning; having this option enabled without knowing it is enabled may be dangerous, correct?